src/modules/Host.rb

```diff
@@ -354,14 +354,25 @@ def set_names(address, names)
   def load_hosts
     return false if SCR.Read(path(".target.size"), CFA::Hosts::PATH) <= 0
 
-    @hosts = CFA::Hosts.new
-    @hosts.load
+    ret = true
+    begin
+      @hosts = CFA::Hosts.new
+      @hosts.load
+
+      # save hosts to check for changes later
+      @hosts_init = CFA::Hosts.new
+      @hosts_init.load
+    rescue RuntimeError => error
```

Comment:
Two problems with this:
1. `RuntimeError` is too generic
2. cannot see where the parse error occurred

In detail:

1. How about having an exception class across YaST, `GarbageDataError`, meaning that YaST cannot understand the config files that it's supposed to edit. My point is that we should not let the user proceed and instead have a global rescue, similar to the (500) Internal Error, but labeled (400) "It's Your Fault!!", or "Lost In Translation", or "Cannot Process Such Data".

2. The actual error is like
    > 'Augeas parsing/serializing error: Iterated lens matched less than it should at /usr/share/augeas/lenses/dist/hosts.aug:23.12-.42:' (RuntimeError)

    and a specific problem for Augeas is that the line location refers to the *lens*, or grammar, and not the actual unparseable data :( Line 23 in hosts.aug is

          let lns = ( empty | comment | record ) *

    We probably cannot solve that in this PR. See also <https://github.com/hercules-team/augeas/wiki/Tracking-down-errors>

Reply:
ideally we shoud log content of file and then we can easily reproduce it.